### ISSUE TITLE
iamidentitymapping command does not honor multiple groups

### ISSUE THREAD

**Issue by mikestef9** [weaveworks/eksctl#3539]
Seeing different behavior based on the cli input format

```
eksctl create iamidentitymapping --cluster staging-cluster --arn arn:aws:iam::123456:role/testing --group group1 --group group2 --username admin

- groups:
  - group1
  - group2
  rolearn: arn:aws:iam::123456:role/testing
  username: admin

```

```
eksctl create iamidentitymapping --cluster staging-cluster --arn arn:aws:iam::123456:role/testing --group "group1,group2" --username admin

- groups:
  - group1,group2
  rolearn: arn:aws:iam::123456:role/testing
  username: admin
```

I'd expect the second command to produce the same output as the first

**Versions**
0.43.0

**Comment by aclevername** [comment]
Thanks for opening the issue @mikestef9 

This appears to be caused by the fact we use a `StringVarArray` instead of a `StringSlice` for the flag

https://github.com/spf13/pflag/blob/85dd5c8bc61cfa382fecd072378089d4e856579d/string_array.go#L75-L78
```
// StringArrayVar defines a string flag with specified name, default value, and usage string.
// The argument p points to a []string variable in which to store the values of the multiple flags.
// The value of each argument will not try to be separated by comma. Use a StringSlice for that.
func (f *FlagSet) StringArrayVar(p *[]string, name string, value []string, usage string) {
```

For example we using `StringSlice` for [vpc subnets flag](https://github.com/weaveworks/eksctl/blob/b1221cda49bc357ebc5f3caae054dc4821fbf0d8/pkg/ctl/create/cluster.go#L88) and it works:  `--vpc-private-subnets="subnet-1,subnet-2"`

**Comment by aclevername** [comment]
fix should be out in todays `0.45.0-rc.0`
